Automattic/wp-calypso
# Plan Step Experiment: Featured Benefits Section

#### Proposed Changes

* Implements an experiment to add a "featured benefits" section in the plan header section, as described in [this figma file](https://www.figma.com/file/PgQSlFBUeqwrtO48yKMj3H/Plans%20page%20system?node-id=4094%3A17926&fuid=776319148157391842).

![Screenshot 2022-10-13 at 15-38-42 Create a site — WordPress com](https://user-images.githubusercontent.com/1269602/195569196-a4fc1b95-c109-4e77-847b-763005277256.png)


### Comments on the code structure

I cloned the [feature comparison folder](https://github.com/Automattic/wp-calypso/tree/trunk/client/my-sites/plan-features-comparison) into a new folder and named it `plan-comparison-with-highlights`, so the changes look deceptively big when in fact _most_ of the lines are similar to the ones in `plan-features-comparison` folder. Here are a few important changes you might want to review:

* In `client/my-sites/plan-comparison-with-highlights/index.jsx`, you might want to trace from [this line](https://github.com/Automattic/wp-calypso/pull/68804/files#diff-266d94ba701b33cd3d227e64174db528cef7816eccf0001e1c7cbeb43f74fb3aR109) which makes a call to `this.renderHighlightFeatures()` to render the featured benefits section. The rest of the file is similar to the corresponding file in `plan-features-comparison` folder.
* Layout change in `renderPlansHeaderNoTabs()` in `client/my-sites/plan-comparison-with-highlights/header.jsx` to conform to the UI in the [FIgma file](https://www.figma.com/file/PgQSlFBUeqwrtO48yKMj3H/Plans-page-system?node-id=4094%3A17926).
* Layout change in `client/my-sites/plan-comparison-with-highlights/item.jsx` to render the material icons for the featured benefits section, and retain the gridicons for the plan features section.
* I also made a minor SCSS change in `client/my-sites/plan-features-comparison/style.scss`, so that the plan pill UI is rendered correctly in both control and treatment. 

### Testing Instructions



* Add yourself to `treatment` of `calypso_signup_plans_step_tagline_202210_v1` experiment and verify that the plans step UI matches the Figma file.
* In `treatment`, switch the Monthly/Annual toggle and verify that annual only features are struck off in the featured benefits section.
* Verify that the control version is unaffected.
* Verify that the tailored onboarding flow and launch flow are unaffected when assigned to `treatment` variation. 
*
